### PR TITLE
chore: check stale time duration using utc time

### PR DESCRIPTION
**Pull request by lazpavel**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-provider-awscloudformation/src/zip-util.js [hunk 1]
```diff
@@ -1,56 +0,0 @@
...
-          reject(objectError);
-          return;
-        }
-
-        fs.ensureDirSync(tempDir);
-
-        const buff = Buffer.from(objectResult);
-        const tempfile = `${tempDir}/${zipFileName}`;
-
-        fs.writeFile(tempfile, buff, err => {
```
Comment (lazpavel) [inline packages/amplify-provider-awscloudformation/src/zip-util.js:28]: the fix for flaky delete in notifications-sms-pull.test.ts as sometimes the writeFile was not finishing before the extractZip method was called in https://github.com/aws-amplify/amplify-cli/blob/dev/packages/amplify-provider-awscloudformation/src/delete-env.ts#L58-L59

Comment (jhockett) [reply packages/amplify-provider-awscloudformation/src/zip-util.js:28]: Nice find!

**Review by jhockett — APPROVED** [review]

File: packages/amplify-provider-awscloudformation/src/zip-util.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+/**
+ * Downloads a zip file from S3
+ */
+export const downloadZip = async (s3: S3, tempDir: string, zipFileName: string, envName: string): Promise<string> => {
+  const log = logger('downloadZip.s3.getFile', [{ Key: zipFileName }, envName]);
+  log();
+  const objectResult = await s3.getFile({ Key: zipFileName }, envName);
+  fs.ensureDirSync(tempDir);
+  const buff = Buffer.from(objectResult);
+  const tempFile = `${tempDir}/${zipFileName}`;
```
Comment (jhockett) [inline packages/amplify-provider-awscloudformation/src/zip-util.ts:19]:
```suggestion
  const tempFile = path.join(tempDir, zipFileName);
```

File: packages/amplify-provider-awscloudformation/src/zip-util.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { $TSAny, AmplifyFault } from 'amplify-cli-core';
```
Comment (jhockett) [inline packages/amplify-provider-awscloudformation/src/zip-util.ts:1]: nit: `$TSAny` is unused

**Review by awsluja — APPROVED** [review]

**Pull request merged**